plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteClient.java

```diff
@@ -0,0 +1,595 @@
+        }
+    }
+
+    @Override
+    public JdbcOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        if (tableMetadata.getComment().isPresent()) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables with table comment");
+        }
+
+        SchemaTableName schemaTableName = tableMetadata.getTable();
+        String schema = schemaTableName.getSchemaName();
+        if (!getSchemaNames(session).contains(schema)) {
+            throw new SchemaNotFoundException(schema);
+        }
+
+        int expectedSize = tableMetadata.getColumns().size();
+        ImmutableList.Builder<String> columns = ImmutableList.builderWithExpectedSize(expectedSize);
+        ImmutableList.Builder<String> columnNames = ImmutableList.builderWithExpectedSize(expectedSize);
+        ImmutableList.Builder<Type> columnTypes = ImmutableList.builderWithExpectedSize(expectedSize);
+        for (ColumnMetadata columnMetadata : tableMetadata.getColumns()) {
+            columns.add(getColumnDefinitionSql(session, columnMetadata, columnMetadata.getName()));
+            columnNames.add(columnMetadata.getName());
+            columnTypes.add(columnMetadata.getType());
+        }
+        String sql = buildCreateSql(schemaTableName, columns.build(), tableMetadata.getProperties());
+
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            execute(session, connection, sql);
+
+            return new IgniteOutputTableHandle(
+                    schemaTableName.getSchemaName(),
+                    schemaTableName.getTableName(),
+                    columnNames.build(),
+                    columnTypes.build(),
+                    Optional.empty(),
+                    IgniteTableProperties.getPrimaryKey(tableMetadata.getProperties()).isEmpty() ? Optional.of(IGNITE_DUMMY_ID) : Optional.empty());
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, "Error creating Ignite table", e);
+        }
+    }
+
+    private String buildCreateSql(SchemaTableName schemaTableName, List<String> columns, Map<String, Object> tableProperties)
+    {
+        ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
+        ImmutableList.Builder<String> columnDefinitions = ImmutableList.builder();
+        columnDefinitions.addAll(columns);
+
+        List<String> primaryKeys = IgniteTableProperties.getPrimaryKey(tableProperties);
+        checkArgument(primaryKeys.size() < columns.size(), "Ignite table must have at least one non PRIMARY KEY column.");
+        if (primaryKeys.isEmpty()) {
+            columnDefinitions.add(quoted(IGNITE_DUMMY_ID) + " VARCHAR NOT NULL");
+            primaryKeys = ImmutableList.of(IGNITE_DUMMY_ID);
+        }
+        columnDefinitions.add("PRIMARY KEY (" + join(", ", primaryKeys.stream().map(this::quoted).collect(joining(", "))) + ")");
```

Comment:
Follo-up is probably fine. It would be better to verify the specified `primary_key` columns exist in the column definition. Currently, broken create table will be executed in Ignite.

Reply:
https://github.com/trinodb/trino/issues/16215